code/game/objects/items/stacks/sheets/glass.dm

```diff
@@ -360,6 +360,12 @@
 				H.apply_damage(5, BRUTE, picked_def_zone)
 				H.Weaken(3)
 				if(cooldown < world.time - 10) //cooldown to avoid message spam.
-					H.visible_message("<span class='danger'>[H] steps in the broken glass!</span>", \
-							"<span class='userdanger'>You step in the broken glass!</span>")
-					cooldown = world.time
+					if(!H.incapacitated())
+						H.visible_message("<span class='danger'>[H] steps in the broken glass!</span>", \
+								"<span class='userdanger'>You step in the broken glass!</span>")
+						cooldown = world.time
+					else
+						H.visible_message("<span class='danger'>[H] slides on the broken glass!</span>", \
+								"<span class='userdanger'>You slide on the broken glass!</span>")
+
+						cooldown = world.time
```

Comment:
you don't need the cooldown twice mate.

just put it at the bottom of the if check like so

	if(!H.incapacitated())
		H.visible_message("<span class='danger'>[H] steps in the broken glass!</span>", \
			"<span class='userdanger'>You step in the broken glass!</span>")

	else
		H.visible_message("<span class='danger'>[H] slides on the broken glass!</span>", \
						"<span class='userdanger'>You slide on the broken glass!</span>")

	cooldown = world.time